File: uphsd-cs-Urbino-Carlos/linear-algebra/ assignment-01/DeterminantSolver.java
/**
 * =====================================================
 * Student Name    : Carlos Miguel M. Urbino
 * Course          : Math 101 — Linear Algebra
 * Assignment      : Programming Assignment 1 — 3x3 Matrix Determinant Solver
 * School          : University of Perpetual Help System DALTA, Molino Campus
 * Date            : [DATE COMPLETED]
 * GitHub Repo     : https://github.com/Crlosmiguel/uphsd-cs-urbino-carlos
 *
 * Description:
 *   This program computes the determinant of a hardcoded 3x3 matrix assigned
 *   to Carlos Miguel M. Urbino using cofactor expansion along the first row.
 * =====================================================
 */

public class DeterminantSolver {

    // Assigned matrix
    static int[][] matrix = {
        {3, 5, 4},
        {6, 2, 1},
        {4, 3, 5}
    };

    // 2x2 determinant helper
    static int computeMinor(int a, int b, int c, int d) {
        return (a * d) - (b * c);
    }

    // Print matrix
    static void printMatrix(int[][] m) {
        System.out.println("┌               ┐");
        for (int[] row : m) {
            System.out.printf("│  %2d  %2d  %2d  │%n", row[0], row[1], row[2]);
        }
        System.out.println("└               ┘");
    }

    static void solveDeterminant(int[][] m) {

        System.out.println("===================================================");
        System.out.println("  3x3 MATRIX DETERMINANT SOLVER");
        System.out.println("  Student: Carlos Miguel M. Urbino");
        System.out.println("  Assigned Matrix:");
        System.out.println("===================================================");

        printMatrix(m);

        System.out.println("===================================================");
        System.out.println("Expanding along Row 1 (cofactor expansion):\n");

        int minor11 = computeMinor(m[1][1], m[1][2], m[2][1], m[2][2]);
        System.out.printf(
        "  Step 1 — Minor M₁₁: det([%d,%d],[%d,%d]) = (%d×%d) - (%d×%d) = %d%n",
        m[1][1], m[1][2], m[2][1], m[2][2],
        m[1][1], m[2][2], m[1][2], m[2][1], minor11);

        int minor12 = computeMinor(m[1][0], m[1][2], m[2][0], m[2][2]);
        System.out.printf(
        "  Step 2 — Minor M₁₂: det([%d,%d],[%d,%d]) = (%d×%d) - (%d×%d) = %d%n",
        m[1][0], m[1][2], m[2][0], m[2][2],
        m[1][0], m[2][2], m[1][2], m[2][0], minor12);

        int minor13 = computeMinor(m[1][0], m[1][1], m[2][0], m[2][1]);
        System.out.printf(
        "  Step 3 — Minor M₁₃: det([%d,%d],[%d,%d]) = (%d×%d) - (%d×%d) = %d%n",
        m[1][0], m[1][1], m[2][0], m[2][1],
        m[1][0], m[2][1], m[1][1], m[2][0], minor13);

        int c11 =  m[0][0] * minor11;
        int c12 = -m[0][1] * minor12;
        int c13 =  m[0][2] * minor13;

        System.out.println();
        System.out.printf("  Cofactor C₁₁ = (+1) × %d × %d = %d%n", m[0][0], minor11, c11);
        System.out.printf("  Cofactor C₁₂ = (-1) × %d × %d = %d%n", m[0][1], minor12, c12);
        System.out.printf("  Cofactor C₁₃ = (+1) × %d × %d = %d%n", m[0][2], minor13, c13);

        int det = c11 + c12 + c13;

        System.out.printf("%n  det(M) = %d + (%d) + %d%n", c11, c12, c13);

        System.out.println("===================================================");
        System.out.printf("  ✓  DETERMINANT = %d%n", det);
        System.out.println("===================================================");
    }

    public static void main(String[] args) {
        solveDeterminant(matrix);
    }
}

File: uphsd-cs-Urbino-Carlos/linear-algebra/ assignment-01/determinant_solver.js
/**
 * =====================================================
 * Student Name    : Carlos Miguel M. Urbino
 * Course          : Math 101 — Linear Algebra
 * Assignment      : Programming Assignment 1 — 3x3 Matrix Determinant Solver
 * School          : University of Perpetual Help System DALTA, Molino Campus
 * Date            : [DATE COMPLETED]
 * GitHub Repo     : https://github.com/[your-username]/uphsd-cs-urbino-carlos
 * Runtime         : Node.js
 * =====================================================
 */

const matrix = [
    [3,5,4],
    [6,2,1],
    [4,3,5]
];

function printMatrix(m){
    console.log("┌               ┐");
    m.forEach(r=>{
        console.log(`│  ${r[0]}  ${r[1]}  ${r[2]}  │`);
    });
    console.log("└               ┘");
}

function computeMinor(a,b,c,d){
    return (a*d)-(b*c);
}

function solveDeterminant(m){

    const line="=".repeat(52);

    console.log(line);
    console.log("  3x3 MATRIX DETERMINANT SOLVER");
    console.log("  Student: Carlos Miguel M. Urbino");
    console.log("  Assigned Matrix:");
    console.log(line);

    printMatrix(m);

    console.log(line);
    console.log("\nExpanding along Row 1 (cofactor expansion):\n");

    const minor11 = computeMinor(m[1][1],m[1][2],m[2][1],m[2][2]);
    console.log(
    `  Step 1 — Minor M₁₁: det([${m[1][1]},${m[1][2]}],[${m[2][1]},${m[2][2]}]) = ${minor11}`
    );

    const minor12 = computeMinor(m[1][0],m[1][2],m[2][0],m[2][2]);
    console.log(
    `  Step 2 — Minor M₁₂: det([${m[1][0]},${m[1][2]}],[${m[2][0]},${m[2][2]}]) = ${minor12}`
    );

    const minor13 = computeMinor(m[1][0],m[1][1],m[2][0],m[2][1]);
    console.log(
    `  Step 3 — Minor M₁₃: det([${m[1][0]},${m[1][1]}],[${m[2][0]},${m[2][1]}]) = ${minor13}`
    );

    const c11 = m[0][0]*minor11;
    const c12 = -m[0][1]*minor12;
    const c13 = m[0][2]*minor13;

    console.log();
    console.log(`  Cofactor C₁₁ = (+1) × ${m[0][0]} × ${minor11} = ${c11}`);
    console.log(`  Cofactor C₁₂ = (-1) × ${m[0][1]} × ${minor12} = ${c12}`);
    console.log(`  Cofactor C₁₃ = (+1) × ${m[0][2]} × ${minor13} = ${c13}`);

    const det = c11 + c12 + c13;

    console.log();
    console.log(`  det(M) = ${c11} + (${c12}) + ${c13}`);
    console.log(line);
    console.log(`  ✓  DETERMINANT = ${det}`);
    console.log(line);
}

solveDeterminant(matrix);